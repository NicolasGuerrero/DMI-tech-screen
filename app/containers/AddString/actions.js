/*
 *
 * AddString actions
 *
 */

import {
  CHANGE_STRING_INPUT,
  ADD_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_ERROR,
} from './constants';

export function changeStringInput(newString) {
  console.log('string from actions', newString);
  return {
    type: CHANGE_STRING_INPUT,
    newString,
  };
}

export function addString() {
  return {
    type: ADD_STRING,
  };
}

export function stringAdded(string) {
  return {
    type: ADD_STRING_SUCCESS,
    string,
  };
}

export function stringAddingError(error) {
  return {
    type: ADD_STRING_ERROR,
    error,
  };
}
