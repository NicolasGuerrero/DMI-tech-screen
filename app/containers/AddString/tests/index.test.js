/**
 *
 * Tests for AddString
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import { AddString, mapDispatchToProps } from '../index';
import { changeStringInput, addString } from '../actions';
import configureStore from '../../../configureStore';

describe('<AddString />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    const reset = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString dispatch={dispatch} reset={reset} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const reset = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString reset={reset} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should not call onSubmitForm if username is an empty string', () => {
    const submitSpy = jest.fn();
    const reset = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString
            onChangeString={() => {}}
            onSubmitForm={submitSpy}
            reset={reset}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  it('should not call onSubmitForm if username is null', () => {
    const submitSpy = jest.fn();
    const reset = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <AddString
            newString=""
            onChangeString={() => {}}
            onSubmitForm={submitSpy}
            reset={reset}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(submitSpy).not.toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeString', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeString).toBeDefined();
      });

      it('should dispatch changeUsername when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        const newString = 'Test String.';
        result.onChangeString({ target: { value: newString } });
        expect(dispatch).toHaveBeenCalledWith(changeStringInput(newString));
      });
    });

    describe('onSubmitForm', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onSubmitForm).toBeDefined();
      });

      it('should dispatch addString when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onSubmitForm();
        expect(dispatch).toHaveBeenCalledWith(addString());
      });

      it('should preventDefault if called with event', () => {
        const preventDefault = jest.fn();
        const result = mapDispatchToProps(() => {});
        const evt = { preventDefault };
        result.onSubmitForm(evt);
        expect(preventDefault).toHaveBeenCalledWith();
      });
    });
  });
});
