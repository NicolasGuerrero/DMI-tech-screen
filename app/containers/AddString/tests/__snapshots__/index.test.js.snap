// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddString /> should render and match the snapshot 1`] = `
<div>
  <form
    class="Form-sc-148ipml-0 hVyDgC"
  >
    <label
      for="newString"
    >
      <input
        class="Input-i9rmlz-0 caCtWU"
        id="newString"
        placeholder="Enter a string..."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </label>
  </form>
  <span
    class="Notification-sc-1hp5mlj-0 jVPNLJ"
    color="red"
  >
    Error: Please try again
  </span>
</div>
`;
